camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+      CamusHourlySweeper.this.metrics.mrSubmitTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+      job.submit();
+      runningJobs.add(job);
+
+      CamusHourlySweeper.this.metrics.mrStartRunningTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+
+      LOG.info("job running for: " + job.getConfiguration().get(TOPIC_AND_HOUR) + ", url: " + job.getTrackingURL());
+      job.waitForCompletion(false);
+      if (!job.isSuccessful()) {
+        throw new RuntimeException("hadoop job failed.");
+      }
+    }
+
+    private void setSplitSizes(long targetSplitSize) {
+      LOG.info("Setting target split size " + targetSplitSize);
+      job.getConfiguration().setLong(MAPRED_MAX_SPLIT_SIZE, targetSplitSize);
+      job.getConfiguration().setLong(MAPRED_MIN_SPLIT_SIZE, targetSplitSize);
+    }
+
+    private void determineAndSetNumOfReducers(int numTasks) {
+      int numReducers;
+      if (job.getConfiguration().get(REDUCER_COUNT) != null) {
+        numReducers = job.getConfiguration().getInt(REDUCER_COUNT, DEFAULT_REDUCER_COUNT);
+      } else {
+        numReducers = numTasks;
+      }
+      job.setNumReduceTasks(numReducers);
+    }
+
+    private long getInputSize() throws IOException {
+      long inputSize = 0;
+      for (Path p : inputPaths) {
+        LOG.info("inputPath: " + p.toString() + ", size=" + fs.getContentSummary(p).getLength());
+        inputSize += fs.getContentSummary(p).getLength();
+      }
+      return inputSize;
+    }
+
+    private void addInputAndOutputPathsToFileInputFormat() throws IOException {
+      for (Path path : inputPaths) {
+        FileInputFormat.addInputPath(job, path);
+        FileOutputFormat.setOutputPath(job, tmpPath);
+      }
+    }
+  }
+
+  private class OutlierCollectorRunner implements Runnable {
+
+    private final Properties props;
+    private final FileSystem fs;
+
+    public OutlierCollectorRunner(Properties props, FileSystem fs) {
+      this.props = props;
+      this.fs = fs;
+    }
+
+    @Override
+    public void run() {
+      String inputPaths = this.props.getProperty(CamusHourlySweeper.INPUT_PATHS);
+      String outputPathStr = this.props.getProperty(CamusHourlySweeper.DEST_PATH);
+      Path outputPath = new Path(outputPathStr, "outlier");
+
+      try {
+        fs.mkdirs(outputPath);
+        long destinationModTime = fs.getFileStatus(new Path(outputPathStr)).getModificationTime();
+
+        for (String inputPathStr : inputPaths.split(",")) {
+          Path inputPath = new Path(inputPathStr);
+          for (FileStatus status : fs.globStatus(new Path(inputPath, "*"), new HiddenFilter())) {
+            if (status.getModificationTime() > destinationModTime) {
+              LOG.info("moving " + status.getPath() + " to " + outputPath);
+              fs.rename(status.getPath(), new Path(outputPath, status.getPath().getName()));
+            }
+          }
+        }
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
You shouldn't throw RuntimExceptions from a Runnable. I believe the ExecutorService will just log the error, but won't fail and the process will continue on. Instead you should use the Callable interface, which can throw checked exceptions and can return an exit value.

http://stackoverflow.com/questions/141284/the-difference-between-the-runnable-and-callable-interfaces-in-java

Also, to check for errors thrown by runnable / callable you need to query the Future object:

http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html#get%28%29